src/libraries/Microsoft.Extensions.HostFactoryResolver/src/HostFactoryResolver.cs

```diff
@@ -247,7 +247,7 @@ public object CreateHost()
                         // build to throw
                         _hostTcs.TrySetException(new InvalidOperationException("The entry point exited without ever building an IHost."));
                     }
-                    catch (TargetInvocationException tie) when (tie.InnerException is HostAbortedException)
+                    catch (TargetInvocationException tie) when (tie.InnerException?.GetType().Name == "HostAbortedException")
```

Comment:
nit: Should this instead check the `Exception`'s full name❔

Reply:
The private exception purposefully has a different full name (since it is nested). So we would need to check 2 different "full names".

Even full names wouldn't prevent some other code running in a Host "Build()" sequence naming their exception `Microsoft.Extensions.Hosting.HostAbortedException`. So it isn't a guaranteed solution either.

If, for some reason, someone else names their exception `HostAbortedException`, and it isn't the one we threw below, the tooling will get `null` back from `HostFactoryResolver.ResolveServiceProviderFactory`, and it will handle it just like any other case where the IServiceProvider can't be found (usually displaying a message to the user saying something went wrong).